ADR/0047-caching-for-container-build-layers.md

```diff
@@ -0,0 +1,33 @@
+# 47. Caching for container base images used during builds
+
+Date: 2025-06-22
+
+## Status
+
+Accepted
+
+## Context
+
+Konflux builds container images using tools like Buildah. During the build process, base images are fetched from container registries like `quay.io`. This repeated fetching of the same images results in high network traffic, slower build times, and exposes builds to failures caused by network issues or registry rate-limiting. This negatively impacts both the user experience (in terms of build speed and reliability) and operational costs of running the platform.
+
+## Decision
+
+We will implement a caching layer for container base images used during the build process using Squid HTTP proxies to mitigate these issues.
+
+* **Deployment:** The proxies will be deployed via a configurable Helm chart into a dedicated `proxy` namespace. A minimal, non-caching "passthrough" configuration will be part of the konflux-ci/konflux-ci "core" deployment to ensure systems are not prone to failure or security issues due to the proxy being unavalable.
```

Comment:
Will the proxy be an optional deployment?

Reply:
as this paragraph states, once we crate the option for pipeline tasks to need a proxy, not having a proxy in the cluster may cause failures, or worse, security issues, so not having it at all is not going to be an option, but by default its not going to do any caching (passthrough).

Reply:
I still doubt that we need to have proxy deployment for all Konflux installations even it it's noop.
If we add a parameter to `buildah` task, then it should be opt-in (if you trust proxy endpoint) without any risk. Have I missed something?

Reply:
The ADR states that the proxy will be included in konflux-ci/konflux-ci repo, installed by default, but with a pass-through no-cache configuration. It's presence there (even with that configuration) implies that it's a part of konflux core (or a dependency of konflux core).

Reply:
From @konflux-ci/konflux-governance-committee discussion, we think the squid proxy deployment should **not** be included in konflux-ci/konflux-ci core (it _can_ appear there as an optional add-on, like image controller does).

A downstream administrator that doesn't want to run the proxy should reference build pipelines in their build-service configmap that have the proxy value unset so that their users' builds don't attempt to access it. If a user sets the param in the build pipeline, it should fail, of course, since there's no proxy deployed there.

Reply:
Thanks - can you get that change in before September 10th?

Let me check what you mean: do you mean that the upstream konflux-ci/konflux-ci install `deploy-deps.sh` and `deploy-konflux.sh` scripts will _not_ install the proxy. i.e., someone needs to opt-in by explicitly installing it with an additional invocation of some other script?

Reply:
I've pushed updates as asked, turns out I already made them locally a while ago but forgot to push